src/test/java/com/thoughtworks/gauge/registry/StepRegistryTest.java

```diff
@@ -122,8 +136,27 @@ public void TestIsFileCached() {
         assertFalse(stepRegistry.isFileCached("/someotherpath/file2.java"));
     }
 
+    @Test
+    public void testGetForCurrentProject() {
+        stepRegistry.addStepImplementation(stepValue1, method2, false);
+
+        StepRegistryEntry entry = stepRegistry.getForCurrentProject(stepValue1.getStepText(), method2);
+
+        assertNotNull(entry);
+        assertEquals("com.thoughtworks.gauge.TestStepImplClass.helloWorldWithOneParam", entry.getFullyQualifiedName());
+    }
+
+    @Test
+    public void testGetForCurrentProjectIgnoresExternal() {
+        stepRegistry.addStepImplementation(stepValue1, method1, true);
```

Comment:
I think `stepRegistry.addStepImplementation(stepValue1, method1, true);` should be `stepRegistry.addStepImplementation(stepValue1, method2, true);`

Reply:
no actually the test checks that the stepRegistry does not return a method without matching name, even though the steptext matches